### PR TITLE
Reset SNS pref to re-opt in with updated interstitial

### PR DESCRIPTION
**Pull request by yrliou**
Resolves https://github.com/brave/brave-browser/issues/34700

sns_interstitial_screenshot

Help page added at: https://github.com/brave/brave-browser/wiki/Resolve-Methods-for-Solana-Name-Service

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Visit onybose.sol should see the updated interstitial.

### CODE REVIEW

File: components/resources/decentralized_dns_strings.grdp [hunk 1]
```diff
@@ -40,16 +40,16 @@
...
   <message name="IDS_DECENTRALIZED_DNS_OPT_IN_PRIMARY_BUTTON" desc="The text for the decentralized DNS opt-in interstitial primary button.">
     Proceed using Infura server
   </message>
   <message name="IDS_DECENTRALIZED_DNS_OPT_IN_PRIMARY_INFURA_BUTTON" desc="The text for the decentralized DNS opt-in interstitial primary button for Infura.">
     Proceed using Infura server
   </message>
-  <message name="IDS_DECENTRALIZED_DNS_OPT_IN_PRIMARY_SYNDICA_BUTTON" desc="The text for the decentralized DNS opt-in interstitial primary button for Syndica.">
-    Proceed using Syndica server
+  <message name="IDS_DECENTRALIZED_DNS_OPT_IN_PRIMARY_SNS_BUTTON" desc="The text for the decentralized DNS opt-in interstitial primary button for SNS server.">
+    Proceed using an SNS server
```
Comment (supermassive) [inline components/resources/decentralized_dns_strings.grdp:52]: Maybe `Proceed using a third-party SNS server` or `Proceed using a third-party server` ?
So it matches replacement of `Syndica` with `third-party` above

**Review by supermassive — APPROVED** [review]

File: components/resources/decentralized_dns_strings.grdp (hunk 1 above)
Comment (yrliou) [reply components/resources/decentralized_dns_strings.grdp:52]: https://github.com/brave/devops/issues/10126#issuecomment-1850182954 was just using the exact wording here. :)

**Review by ShivanKaul — APPROVED** [review]
Screenshot and text LGTM.

**Comment by ShivanKaul** [comment]
@yrliou I tweaked https://github.com/brave/brave-browser/wiki/Resolve-Methods-for-Solana-Name-Service/_compare/7a4f4f07ca79a713b5c331f4cb4f6a95439906d5...bec16620c378cb0df8b111c13377ba29b2324138, please take a quick look.

**Comment by yrliou** [comment]
@ShivanKaul The tweak looks to me, thanks!

**Review by mkarolin — APPROVED** [review]
`strings`++

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.63.53 Chromium: 120.0.6099.115 (Official Build) nightly (64-bit)
-- | --
Revision | 6bc3ebcb9c34235e7b910334f3733146b8eaa7e9
OS | Windows 11 Version 22H2 (Build 22621.2861)
```

* ensured that visiting https://onybose.sol loads the correct interstitial page mentioned via  https://github.com/brave/brave-core/pull/21353#issue-2038723510
* ensured clicking on `help page` opens https://github.com/brave/brave-browser/wiki/Resolve-Methods-for-Solana-Name-Service
* ensured that clicking on `Disable` loads the `This site can’t be reached` page
  * ensured that `Resolve Solana Name Service (SNS) domain names` is set as `Disabled` via `brave://settings/web3`
  * ensured that changing it to `Ask` loads the correct interstitial page
* ensured that clicking on `Proceed using SNS server` correctly resolves https://anirudha.co without any issues
  * ensured that `Resolve Solana Name Service (SNS) domain names` is set as `Enabled` via `brave://settings/web3`
  
**`Enabled`** | **`Enabled`** | **`Disabled`** | **`Disabled`** 
----------------|---------------|-----------------|----------------
image | image | image | image

**Pull request merged**